sdk/log/DESIGN.md

```diff
@@ -5,13 +5,27 @@
 `go.opentelemetry.io/otel/sdk/log` provides Logs SDK compliant with the
 [specification](https://opentelemetry.io/docs/specs/otel/logs/sdk/).
 
-The main and recommended use case is to configure the SDK to use an OTLP
-exporter with a batch processor.[^1] Therefore, the design aims to be
-high-performant in this scenario.
-
 The prototype was created in
 [#4955](https://github.com/open-telemetry/opentelemetry-go/pull/4955).
 
+## Background
+
+The goal is to design the exported API of the SDK would have low performance
+overhead. Most importantly, have a design that reduces the amount of heap
+allocations and even make it possible to have a zero-allocation implementation.
+Eliminating the amount of heap allocations reduces the GC pressure which can
+produce some of the largest improvements in performance.[^1]
+
+The main and recommended use case is to configure the SDK to use an OTLP
+exporter with a batch processor.[^2] Therefore, the implementation aims to be
+high-performant in this scenario. Some users that require high throuput may also
+want to use e.g. an [LLTng](https://lttng.org/docs/v2.13/#doc-tracing-your-own-user-application)
+or [ETW](https://learn.microsoft.com/en-us/windows/win32/etw/about-event-tracing)
+exporter with a simple processor. Users may also want to use
```

Comment:
OTel spec defined SimpleProcessor won't scale, as Exports are queued one after the other. In OTel .NET,C++,Rust, we use a custom Processor, which calls the exporter.Export() method on every OnEnd/OnEmit call synchronously. i.e multiple Exports occur in parallel.

Reply:
> OTel spec defined SimpleProcessor won't scale, as Exports are queued one after the other

Where is it defined like this?

https://github.com/open-telemetry/opentele> OTel spec defined SimpleProcessor won't scale, as Exports are queued one after the other

Where is it defined like this?

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/sdk.md#simple-processor

> we use a custom Processor, which calls the exporter.Export() method on every OnEnd/OnEmit call synchronously. i.e multiple Exports occur in parallel.

This is how simple processor is currently implemented in Go Logs SDK.metry-specification/blob/main/specification/logs/sdk.md#simple-processor

> we use a custom Processor, which calls the exporter.Export() method on every OnEnd/OnEmit call synchronously. i.e multiple Exports occur in parallel.

This is how simple processor is currently implemented in Go Logs SDK.

Reply:
> OTel spec defined SimpleProcessor won't scale, as Exports are queued one after the other

Where is it defined like this?

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/sdk.md#simple-processor

> we use a custom Processor, which calls the exporter.Export() method on every OnEnd/OnEmit call synchronously. i.e multiple Exports occur in parallel.

This is how simple processor is currently implemented in Go Logs SDK.

Reply:
I was searching for that wording, but turns out it is not part of the Processor section, but under Exporter spec: 
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/sdk.md#export
_"Export will never be called concurrently for the same exporter instance."_ 

I guess this translates to Processors must queue up requests, so that Exporter.Export() methods are called after previous one returns..

Reply:
Thanks.

This would require adding logic to Exporter...

The language is not normative.  And on the other hand we have 

> Each implementation MUST document the concurrency characteristics the SDK requires of the exporter.

that uses normative language

Reply:
After reading the spec wording for all signals, I am not sure if it was intended in the spec that export() be called concurrently. Also, not sure if the following was not made in normative language intentionally or just missed to call it out.
_"Export will never be called concurrently for the same exporter instance."_

Anyway, given the way Simple processor is implemented in OTel Go, the wording in this PR makes sense. OTel Java also has same implementation of SimpleProcessors.
(OTel Rust, C++, .NET implements SimpleProcessors in a way that only one export() call is made concurrently. Implementing wise, SimpleProcessor takes a lock before invoking export() in these languages).

Reply:
I created https://github.com/open-telemetry/opentelemetry-specification/issues/4134